app/src/main/java/com/doctor/yumyum/presentation/ui/myrecipe/MyRecipeViewModel.kt

```diff
@@ -1,32 +1,74 @@
 package com.doctor.yumyum.presentation.ui.myrecipe
 
+import android.util.Log
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import com.doctor.yumyum.R
 import com.doctor.yumyum.common.base.BaseViewModel
+import com.doctor.yumyum.common.utils.MineFoodType
 import com.doctor.yumyum.data.model.BestRecipe
+import com.doctor.yumyum.data.model.RecipeModel
+import com.doctor.yumyum.data.model.WriteRecipe
+import com.doctor.yumyum.data.remote.response.SearchRecipeResponse
 import com.doctor.yumyum.data.repository.MainRepositoryImpl
+import com.doctor.yumyum.data.repository.MyRecipeRepositoryImpl
+import okhttp3.ResponseBody
+import retrofit2.Response
+import kotlin.coroutines.coroutineContext
 
 class MyRecipeViewModel : BaseViewModel() {
     private val repository = MainRepositoryImpl()
+    private val myRecipeRepository = MyRecipeRepositoryImpl()
 
     private val _mode: MutableLiveData<Int> = MutableLiveData(repository.getMode())
     val mode: LiveData<Int>
         get() = _mode
 
-    private val _bestRecipeList : MutableLiveData<ArrayList<BestRecipe>> = MutableLiveData()
-    val bestRecipeList : LiveData<ArrayList<BestRecipe>>
+    private val _foodType: MutableLiveData<String> = MutableLiveData(MineFoodType.MYFOOD.name)
+    val foodType: LiveData<String>
+        get() = _foodType
+
+    private val _bestRecipeList: MutableLiveData<ArrayList<BestRecipe>> = MutableLiveData()
+    val bestRecipeList: LiveData<ArrayList<BestRecipe>>
         get() = _bestRecipeList
 
-    private val _myRecipeList : MutableLiveData<ArrayList<String>> = MutableLiveData()
-    val myRecipeList : LiveData<ArrayList<String>>
+    private val _myRecipeList: MutableLiveData<ArrayList<RecipeModel>> = MutableLiveData()
+    val myRecipeList: LiveData<ArrayList<RecipeModel>>
         get() = _myRecipeList
 
     fun changeMode() {
         _mode.value =
             if (mode.value == R.string.common_food) R.string.common_beverage else R.string.common_food
-
-        // 현재 모드 저장
         repository.setMode(mode.value ?: R.string.common_food)
     }
-}
\ No newline at end of file
+
+    fun changeFoodType(foodType : String) {
+        _foodType.value = foodType
+    }
+
+
+    suspend fun getMyRecipe(categoryName: String, mineFoodType: String) {
+        try {
+            val response =
+                myRecipeRepository.getMyRecipe(
+                    categoryName = categoryName,
+                    flavors = "",
+                    tags = "",
+                    offset = 0,
+                    pageSize = 10,
+                    mineFoodType = mineFoodType,
+                    minPrice = null,
+                    maxPrice = null,
+                    firstSearchTime = "2022-12-20T12:12:12",
+                    sort = "id",
+                    order = "asc",
+                    status = ""
+                )
+            _myRecipeList.postValue(response.body()?.foods)
```

Comment:
이 부분 response 실패하면 myRecipeList에 null 들어가서 오류 날 것 같습니당 null 체크 해주거나 success인지 체크해줘야 할 것 같습니다
```suggestion
            response.body()?.foods?.let { _myRecipeList.postValue(it) }
```